File: SoftUniCourses/JavaScript/VSCode/JavaScript/whileExe/walking.js
function walking(input) {

    let target = 10000;
    let sum = 0;
    let i = 0;
    let command = input[i];
    let differenceSteps =0;
    
    while (command!== "Going home") {
        let steps = Number(command);
        sum = sum + steps;

        if (sum >= target) {
            console.log("Goal reached! Good job!");
             differenceSteps = sum - target;
            console.log(`${differenceSteps} steps over the goal!`);
        }
        i++;
        command = input[i];
    }
    if (command === "Going home") {
        let stepsToHome = Number(input[i + 1]);
        sum = sum + stepsToHome;

        if (sum >= target) {
            console.log("Goal reached! Good job!");
             differenceSteps = sum - target;
            console.log(`${differenceSteps} steps over the goal!`);
        } else {
            differenceSteps = target - sum;
            console.log(`${differenceSteps} more steps to reach goal.`);
        }
    }
}
walking(["1500","3000","250","1548","2000","Going home","2000"]);

File: SoftUniCourses/SoftUniBasics/whileloops/Walking.java
package whileloops;

import java.util.Scanner;

public class Walking {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        //два варианта за стоп-стъпки >=10000 или команда ..
        //команда или число/бр.стъпки/
        int totalStep = 0;
        String commanda = scanner.nextLine();
        while (!commanda.equals("Going home")){

            int step = Integer.parseInt(commanda);
            totalStep += step;
            if (totalStep>= 10000){
                break;
            }

            commanda = scanner.nextLine();


        }
        if (commanda.equals("Going home")){
            int stepHome = Integer.parseInt(scanner.nextLine());
            totalStep += stepHome;
        }
        if (totalStep>= 10000){
            System.out.println("Goal reached! Good job!");
            int moreSteps = totalStep - 10000;
            System.out.printf("%d steps over the goal!", moreSteps);

        }else {
            int diff = 10000 - totalStep;
            System.out.printf("%d more steps to reach goal.", diff);
        }
    }
}
